src/app/api/tasks/tasks.service.ts

```diff
@@ -0,0 +1,96 @@
+import { CreateTaskRequest, UpdateTaskRequest } from '@/types/dto/tasks.dto'
+import { BaseService } from '@api/core/services/base.service'
+import { Resource } from '@api/core/types/api'
+import { PoliciesService } from '@api/core/services/policies.service'
+import { AssigneeType } from '@prisma/client'
+
+type FilterByAssigneeId = {
+  assigneeId: string
+  assigneeType: AssigneeType
+}
+
+export class TasksService extends BaseService {
+  private buildReadFilters() {
+    const user = this.user
+
+    let filters = { where: { workspaceId: user.workspaceId, OR: undefined as unknown as FilterByAssigneeId[] } }
+
+    if (user.clientId) {
+      filters = {
+        where: {
+          ...filters.where,
+          OR: [{ assigneeId: user.clientId as string, assigneeType: 'client' }],
+        },
+      }
+    }
+    if (user.clientId && user.companyId) {
+      filters = {
+        where: {
+          ...filters.where,
+          OR: [
+            { assigneeId: user.clientId as string, assigneeType: 'client' },
+            { assigneeId: user.companyId, assigneeType: 'company' },
+          ],
+        },
+      }
+    }
+
+    return filters
+  }
+  async getAllTasks() {
+    const user = this.user
+    new PoliciesService(user).authorize('read', Resource.Tasks)
```

Comment:
what is this line doing?

Reply:
@makeitraina I'm going with a resource-level authorization approach. 

Since all controllers use the service layer to access resources, each service method has a line on top using `PoliciesService#authorize` to check if user can perform the need action (read, write, update, etc) on the resource this particular service is associated with.